packages/Webkul/Category/src/Http/Controllers/CategoryController.php

```diff
@@ -195,10 +191,6 @@ public function massDestroy()
                         $suppressFlash = true;
                         Event::dispatch('catalog.category.delete.before', $categoryId);
 
-                        if ($category->products->count() > 0) {
-                            $category->products()->delete();
-                        }
-
                         $category->delete();
 
                         Event::dispatch('catalog.category.delete.after', $categoryId);
```

Comment:
Pass category model object to this event

Reply:
Mentioned changes have been done.